impeller/display_list/aiks_dl_basic_unittests.cc

```diff
@@ -1067,5 +1069,445 @@ TEST_P(AiksTest, EmptySaveLayerRendersWithClear) {
+                                        720 + i * 20, 220 + j * 20),
+                       i * 10, j * 10, paint);
+  }
+  matrix = SkMatrix::Translate(800, 300);
+  paint.setColor(DlColor::kWhite().modulateOpacity(0.5));
+  paint.setColorSource(std::make_shared<DlImageColorSource>(
+      texture, DlTileMode::kRepeat, DlTileMode::kRepeat,
+      DlImageSampling::kMipmapLinear, &matrix));
+
+  draw_rrect_as_path(SkRect::MakeLTRB(800, 410, 1000, 490), 40, 40, paint);
+  draw_rrect_as_path(SkRect::MakeLTRB(860, 350, 940, 550), 40, 40, paint);
+
+  ASSERT_TRUE(OpenPlaygroundHere(builder.Build()));
+}
+
+TEST_P(AiksTest, CoverageOriginShouldBeAccountedForInSubpasses) {
+  auto callback = [&]() -> sk_sp<DisplayList> {
+    DisplayListBuilder builder;
+    builder.Scale(GetContentScale().x, GetContentScale().y);
+
+    DlPaint alpha;
+    alpha.setColor(DlColor::kRed().modulateOpacity(0.5));
+
+    auto current = Point{25, 25};
+    const auto offset = Point{25, 25};
+    const auto size = Size(100, 100);
+
+    static PlaygroundPoint point_a(Point(40, 40), 10, Color::White());
+    static PlaygroundPoint point_b(Point(160, 160), 10, Color::White());
+    auto [b0, b1] = DrawPlaygroundLine(point_a, point_b);
+    SkRect bounds = SkRect::MakeLTRB(b0.x, b0.y, b1.x, b1.y);
+
+    DlPaint stroke_paint;
+    stroke_paint.setColor(DlColor::kYellow());
+    stroke_paint.setStrokeWidth(5);
+    stroke_paint.setDrawStyle(DlDrawStyle::kStroke);
+    builder.DrawRect(bounds, stroke_paint);
+
+    builder.SaveLayer(&bounds, &alpha);
+
+    DlPaint paint;
+    paint.setColor(DlColor::kRed());
+    builder.DrawRect(
+        SkRect::MakeXYWH(current.x, current.y, size.width, size.height), paint);
+
+    paint.setColor(DlColor::kGreen());
+    current += offset;
+    builder.DrawRect(
+        SkRect::MakeXYWH(current.x, current.y, size.width, size.height), paint);
+
+    paint.setColor(DlColor::kBlue());
+    current += offset;
+    builder.DrawRect(
+        SkRect::MakeXYWH(current.x, current.y, size.width, size.height), paint);
+
+    builder.Restore();
+
+    return builder.Build();
+  };
+
+  ASSERT_TRUE(OpenPlaygroundHere(callback));
+}
+
+TEST_P(AiksTest, SaveLayerDrawsBehindSubsequentEntities) {
+  // Compare with https://fiddle.skia.org/c/9e03de8567ffb49e7e83f53b64bcf636
+  DisplayListBuilder builder;
+  DlPaint paint;
+
+  paint.setColor(DlColor::kBlack());
+  SkRect rect = SkRect::MakeXYWH(25, 25, 25, 25);
+  builder.DrawRect(rect, paint);
+
+  builder.Translate(10, 10);
+
+  DlPaint save_paint;
+  builder.SaveLayer(nullptr, &save_paint);
+
+  paint.setColor(DlColor::kGreen());
+  builder.DrawRect(rect, paint);
+
+  builder.Restore();
+
+  builder.Translate(10, 10);
+  paint.setColor(DlColor::kRed());
+  builder.DrawRect(rect, paint);
+
+  ASSERT_TRUE(OpenPlaygroundHere(builder.Build()));
+}
+
+TEST_P(AiksTest, SiblingSaveLayerBoundsAreRespected) {
+  DisplayListBuilder builder;
+  DlPaint paint;
+  SkRect rect = SkRect::MakeXYWH(0, 0, 1000, 1000);
```

Comment:
I don't remember where it caused a problem but in some tool I've used to look at goldens a transparent background with black stuff drawn on it just looks like a giant field of black, so I usually paint a background color in my tests...